modules/Blockchain.js

```diff
@@ -561,10 +561,16 @@ class Blockchain {
         }
 
         const that = this;
+        let processingEvents = false;
+
         const handle = setInterval(async () => {
             if (!that.appState.started) {
                 return;
             }
+            if (processingEvents) {
```

Comment:
I am not sure is it still safe, so I suggest to use more advanced approaches for further implementations, like: https://www.npmjs.com/package/async-mutex